File: backend/src/main/scala/microservices/operations/overall/objects/ManagerRefundDecisionPlannerRequest.scala
// 本文件定义 `ApproveManagerRefundPlanner` 和 `RejectManagerRefundPlanner` 对应的请求对象。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ManagerRefundDecisionPlannerRequest(managerId: String, managerType: String, orderId: String)
object ManagerRefundDecisionPlannerRequest:
  given sourceEncoder: Encoder[ManagerRefundDecisionPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[ManagerRefundDecisionPlannerRequest] = deriveDecoder

File: frontend/src/microservices/operations/overall/objects/ManagerRefundDecisionPlannerRequest.ts
// 本文件定义 `ApproveManagerRefundPlanner` 和 `RejectManagerRefundPlanner` 的请求对象。

export type ManagerRefundDecisionPlannerRequest = {
  managerId: string
  managerType: string
  orderId: string
}

export const managerRefundDecisionPlannerRequestFromJson = (json: string): ManagerRefundDecisionPlannerRequest =>
  JSON.parse(json) as ManagerRefundDecisionPlannerRequest

export const managerRefundDecisionPlannerRequestToJson = (value: ManagerRefundDecisionPlannerRequest): string =>
  JSON.stringify(value)
